### PR TITLE
[DPE-3086] Improve CI/CD pipeline

### PR DESCRIPTION
**Pull request by deusebio**
I noticed that in Kafka CI/CD pipeline, we have been currently publishing wrong tags, that are not according to the standard provided as company. 

While fixing this, I also provided some upgrades we did in Spark rock image to: 

* make sure that versions and release branching names are consistent
* add extra labels to the image such that we can track a bit better which code originated the image

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -5,49 +5,110 @@ on:
...
+            VERSION_TAG="${{ needs.release_checks.outputs.version }}-${{ needs.release_checks.outputs.base }}_edge"
+            
+            docker tag ${IMAGE_NAME}:${TAG} ${IMAGE_NAME}:${VERSION_TAG}
+            
+            echo "Publishing ${IMAGE_NAME}:${VERSION_TAG}"          
+            docker push ${IMAGE_NAME}:${VERSION_TAG}
+          fi
+          
+          
+
```
Comment (zmraul) [inline .github/workflows/publish.yaml]: nit: a couple of extra empty lines

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -5,49 +5,110 @@ on:
...
-          risk=edge
-          tag=${version}-${base}_${risk}
-          channel=$(echo ${{ github.ref_name }} | cut -d "-" -f 1)
+      
+          REPOSITORY="ghcr.io/canonical/"
+          RISK=${{ needs.release_checks.outputs.risk }}
+          TRACK=${{ needs.release_checks.outputs.track }}
+          if [ ! -z "$RISK" ] && [ "${RISK}" != "no-risk" ]; then TAG=${TRACK}_${RISK}; else TAG=${TRACK}; fi
+      
+          IMAGE_NAME="${REPOSITORY}charmed-kafka"
```
Comment (zmraul) [inline .github/workflows/publish.yaml]: nit: `REPOSITORY` is not used anywhere else that I see, so it can all be declared here

**Review by zmraul — APPROVED** [review]
LGTM

**Review by zmraul — CHANGES_REQUESTED** [review]
Same [comment in zookeeper](https://github.com/canonical/charmed-zookeeper-rock/pull/26#pullrequestreview-1836761622) can be squeezed on this PR as well

**Review by zmraul — APPROVED** [review]

**Review by marcoppenheimer — APPROVED** [review]

**Pull request merged**